lighthouse-core/lib/median-run.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/** @param {LH.Result} lhr @param {string} auditName */
+const getNumericValue = (lhr, auditName) =>
+  (lhr.audits[auditName] && lhr.audits[auditName].numericValue) || NaN;
+
+/**
+ * @param {LH.Result} lhr
+ * @param {number} medianFcp
+ * @param {number} medianInteractive
+ */
+function getMedianSortValue(lhr, medianFcp, medianInteractive) {
+  const distanceFcp = medianFcp - getNumericValue(lhr, 'first-contentful-paint');
+  const distanceInteractive =
+    medianInteractive - getNumericValue(lhr, 'interactive');
+
+  return distanceFcp * distanceFcp + distanceInteractive * distanceInteractive;
+}
+
+/**
+ * We want the run that's closest to the median of the FCP and the median of the TTI. 
+ * We're using the squared distance for that (https://en.wikipedia.org/wiki/Euclidean_distance).
+ * @param {Array<LH.Result>} runs
+ * @return {LH.Result|undefined}
+ */
+function computeMedianRun(runs) {
```

Comment:
worth pointing out this is *a* type of median, not *the* median :) because there are several ways to naturally  define the median in multiple dimensions, depending on what aspect of the 1D median you want to preserve: https://en.wikipedia.org/wiki/Median#Multivariate_median (this is choosing the observation closest to the [centerpoint](https://en.wikipedia.org/wiki/Centerpoint_(geometry)))

We might try the other medians to see how they look sometime, but I don't think it would ever matter for 3, 5, 9, etc samples

Reply:
> worth pointing out this is a type of median, not the median

The history of this method in LHCI is not even trying to find the median nor do we call it the median but rather a "representative" run :) in CI I still think we'll end up doing something that more closely resembles the min rather than the median eventually, but in core it made more sense to me to be explicit about the median-usage and offer new methods in the future if necessary.